contracts/core/EntryPoint.sol

```diff
@@ -356,9 +381,11 @@ contract EntryPoint is IEntryPoint, StakeManager, NonceManager, ReentrancyGuardT
                 mUserOp.paymasterPostOpGasLimit +
                 INNER_GAS_OVERHEAD
             ) {
+                uint256 gasUsed = preGas - gasleft();
                 assembly ("memory-safe") {
                     mstore(0, INNER_OUT_OF_GAS)
-                    revert(0, 32)
+                    mstore(32, gasUsed)
+                    revert(0, 64)
```

Comment:
If we are making a  big refactoring now, would you consider extracting this entire `unchecked { ... }` block to some `_verifyEnoughGasForInner` function? This is a very distracting code.